repos/system_upgrade/el7toel8/actors/selinux/selinuxprepare/tests/component_test.py

```diff
@@ -0,0 +1,101 @@
+import os
+
+import pytest
+
+from leapp.snactor.fixture import current_actor_context
+from leapp.models import SELinuxModule, SELinuxModules, SELinuxCustom
+from leapp.libraries.stdlib import api, run, CalledProcessError
+from leapp.reporting import Report
+
+enabled = False
+
+test_modules = [
+    ["400", "mock1"],
+    ["99", "mock1"],
+    ["300", "mock1"],
+    ["400", "mock2"],
+    ["999", "mock3"]
+]
+
+semanage_commands = [
+    ['fcontext', '-t', 'httpd_sys_content_t', '"/web(/.*)?"'],
+    ['fcontext', '-t', 'ganesha_var_run_t', '"/ganesha(/.*)?"'],
+    ['port', '-t', 'http_port_t', '-p', 'udp', '81'],
+    ['permissive', 'abrt_t']
+]
+
+# save value of semodule -lfull for comparison
+semodule_lfull = ""
+semanage_export = ""
+
+if enabled:
+    try:
+        semodule = run(["semodule", "-lfull"], split=False)
+        semodule_lfull = semodule.get("stdout", "")
+        semanage = run(["semanage", "export"], split=False)
+        semanage_export = semanage.get("stdout", "")
+    except CalledProcessError as e:
+        api.current_logger().warning("Error listing SELinux customizations: %s", str(e.stderr))
+
+testmoduledir = os.path.join(os.getcwd(), "tests/mock_modules/")
```

Comment:
This way tests are not discovered properly, as `ACTOR=SELinuxPrepare make test` will have leapp-repository/ as cwd, not actor-specific directory.
You may try changing it to `testmoduledir = "tests/mock_modules/"` and constructing mock data dir in test_SELinuxPrepare() like
```
testdir = os.path.join(os.getenv('PYTEST_CURRENT_TEST').rsplit('/', 2)[0], testmoduledir)
run(["semodule", "-X", priority, "-i", os.path.join(testdir, module + ".cil")])
```

After that the dir is right but component test starts to fail with AssertionError
```
>   assert semodule_lfull == semodule.get("stdout", "")
E   AssertionError: assert '100 abrt    ... pp         \n' == '300 mock1    ... pp         \n'
E     + 300 mock1             cil         
E       100 abrt              pp         
E       100 accountsd         pp         
E       100 acct              pp         
E       100 afs               pp         
E       100 aiccu             pp         
E       100 aide              pp         ...
E     
E     ...Full output truncated (400 lines hidden), use '-vv' to show
```

Reply:
Hmm interesting that after rebase the test actually passes. Will look into it tomorrow.

Reply:
It seems that CWD differs in between tests for "make test"... fixed by using path to test file